tests/integration/test_migrate.py

```diff
@@ -2,6 +2,134 @@
 from redis import ResponseError
 
 
+def test_migration_basic(cluster_factory):
+    """
+    Simulates full migration process. Test generates keys in a cluster and moves
+    half of the keys to another cluster.
+    """
+
+    cluster1 = cluster_factory().create(3, raft_args={
+        'sharding': 'yes',
+        'external-sharding': 'yes'})
+    cluster2 = cluster_factory().create(3, raft_args={
+        'sharding': 'yes',
+        'external-sharding': 'yes'})
+
+    key_count = 11500
+
+    # Fill two specific slots with keys. One of these slots will be migrated.
+    for i in range(key_count):
+        cluster1.execute('set', '{key}' + str(i), 'value')   # slot 12539
+        cluster1.execute('set', '{key2}' + str(i), 'value')  # slot 4998
+
```

Comment:
maybe add a validation here on the size: 
```assert cluster1.execute('dbsize') == 2 * key_count```

Reply:
Right. I usually prefer to have guards before and after to catch an issue asap, but as you wish. `Assert`s are not only to validate our changes, but declare an assumption.